keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,832 @@
+
+###### 1. Admin creates BucketAccessClass, User creates BucketAccess
 
-# Object Relationships
+The BucketAccessClass represents a set of common properties shared by multiple BucketClaims. It is used to specify policies for creating access credentials, and also for configuring driver-specific access parameters. More information about BucketAccessClass is [here](#bucketaccessclass)
 
-The diagram below describes the relationships between various resources in the COSI ecosystem. 
+The BucketAccess is used to request access to a bucket. It contains fields for choosing the Bucket for which the credentials will be generated, and also includes a bucketAccessClassName field, which in-turn contains configuration for authorizing users to access buckets. More information about BucketAccess is [here](#bucketaccess)
+
+BucketAccessClass can be used to specify a authorization mechanism. It can be one of 
+ - KEY  (__default__) 
+ - IAM 
+
+The KEY based mechanism is where access and secret keys are generated to be provided to pods. IAM style is where pods are implicitly granted access to buckets by means of a metadata service. IAM style access provides greater control for the infra/cluster administrator to rotate secret tokens, revoke access, change authorizations etc., which makes it more secure.
+
+```
+    BucketAccess - ba1                                        BucketAccessClass - bac1
+    |---------------------------------------|                 |----------------------------------|
+    | metadata:                             |                 | provisioner: s3.amazonaws.com    |
+    |   namespace: ns1                      |                 | parameters:                      |
+    | spec:                                 |                 |   key: value                     |
+    |   bucketAccessClassName: bac1         |                 | authenticationType: KEY          |
+    |   bucketClaimName: bcl1               |                 |----------------------------------|
+    |   credentialsSecretName: bucketcreds1 |
+    | status:                               |
+    |   conditions:                         |
+    |   - name: AccessGranted               |
+    |     value: False                      |
+    |---------------------------------------|
+```
 
-![Object Relationships](images/object-rel.png)
+The `credentialsSecretName` is the name of the secret that COSI will generate containing  credentials to access the bucket. The same secret name has to be set in the podSpec as well as the projected secret volumes.
 
-# Workflows
-Here we describe the workflows used to automate provisioning of new and existing buckets, and the de-provisioning of these buckets.
+In case of IAM style authentication, along with the `credentialsSecretName`, `serviceAccountName` field must also be specified. This will map the specified serviceaccount to the appropriate service account in the OSP.
```

Comment:
The workflows focus a lot on workloads, but do we also want to support users? ie, instead of granting access to a service account, can we also grant access to a user?

Reply:
The scope of access control is namespace level. Any user that has access to a bucket can only utilize it by running a pod/workload with credentials authorizing them. Since all forms of credential specifications (Secret, EnvVar, etc...) are accessible to all pods in the namespace anyways, there is no need to control access at a smaller level of granularity.

TL;DR Users cannot access a bucket without opening it upto other pods in the same namespace. Therefore, namespace level access control is enough.